backend/models.py

```diff
@@ -76,6 +88,9 @@ class Task(models.Model):
     def get_absolute_url(self):
         return reverse("task-detail", args=[str(self.id)])
 
+    def __str__(self):
+        return self.data # should be "{User} task"

```

Comment:
getting associated user string should be `self.user.get_username()` or `self.user.get_full_name()` if i'm not wrong

Reply:
also task data may not be trivial (e.g. the result of a task might be like a 5 megabyte log file), so what makes more sense as the `__str__` is probably something like 
```python
return f"Task (id={self.id}, user={self.user.get_username()}, endpoint={self.endpoint}, start_time={self.start_time}, ...)"
```
not sure if there's a django library that autogenerates `__str__` definitions like how `dataclass` or `attrs` do it, but typically usually what you want as either the `__str__` or the `__repr__`

personally i prefer putting this stuff in `__str__` since that's what you'll be seeing when you inevitably print objects out for debugging; if you need more user-friendly names for stuff, just define a function like 
```python
def get_human_repr(self):
    return f"Task #{self.id} ({self.user.get_username()})"
```
or whatever